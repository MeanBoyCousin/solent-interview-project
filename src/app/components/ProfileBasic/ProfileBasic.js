import React from 'react'

import { mediumXInOut } from 'Motion/variants'
import { ProfileWrapper } from './ProfileBasic.styled'

const ProfileBasic = ({ picture, name, uuid, setModal }) => {
    return (
        <ProfileWrapper
            variants={mediumXInOut}
            initial="hidden"
            animate="visible"
            exit="exit"
            onClick={() => {
                setModal({ visible: true, uuid: uuid })
            }}
        >
            {picture && name ? (
                <>
                    <img
                        className="thumbnail"
                        src={picture}
                        alt={`The profile picture for ${name}`}
                    />
                    <p>{name}</p>
                    <img
                        className="arrow"
                        src={`${process.env.ICONS}circle-arrow.svg`}
                        alt="An arrow pointing right enclosed inside a circle"
                    />
                </>
            ) : (
                <>
                    <div
                        data-testid="skeleton"
                        className="thumbnail skeleton skeleton-thumbnail"
                    ></div>
                    <div className="skeleton skeleton-name"></div>
                </>
            )}
        </ProfileWrapper>
    )
}

export { ProfileBasic }
