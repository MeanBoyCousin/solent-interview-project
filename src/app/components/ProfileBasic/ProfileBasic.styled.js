import styled from 'styled-components'

const ProfileWrapper = styled.div`
    position: relative;
    display: flex;
    align-items: center;
    width: 85%;
    margin: 2.5% 0;
    background: white;
    border: 1px solid black;
    box-shadow: ${props => props.theme['shadow-sm']};
    cursor: pointer;
    transition: ${props => props.theme['ease-100']};

    &:hover {
        box-shadow: ${props => props.theme['shadow-md']};
    }

    &:active {
        box-shadow: ${props => props.theme['shadow-lg']};
    }

    .thumbnail {
        width: 48px;
        height: 48px;
        border-radius: 100%;
        border: 1px solid black;
        box-shadow: ${props => props.theme['shadow-sm']};
        margin: 2.5%;
    }

    p {
        font-size: 0.8rem;
        font-weight: 600;
    }

    .arrow {
        position: absolute;
        top: 0;
        right: 0;
        height: 100%;
        transform: scale(0.7);
    }

    .skeleton {
        background: ${props => props.theme['dark-grey']};
        animation: ${props => props.theme.pulse};
    }

    .skeleton-thumbnail {
        width: 48px;
        height: 48px;
    }

    .skeleton-name {
        width: 50%;
        height: 12px;
    }

    @media (min-width: 768px) {
        p {
            font-size: 0.9rem;
        }

        .arrow {
            transform: scale(0.6);
        }
    }

    @media (min-width: 992px) {
        width: 50%;
        margin: 2.5% 0 0 0;

        &:last-child {
            margin: 2.5% 0;
        }

        p {
            font-size: 1rem;
        }

        .arrow {
            transform: scale(0.5);
        }
    }
`

export { ProfileWrapper }
