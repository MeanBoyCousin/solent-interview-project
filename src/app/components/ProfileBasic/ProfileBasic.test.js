import React from 'react'
import 'dotenv/config'
import '@testing-library/jest-dom'
import { render, cleanup } from '@testing-library/react'

import { ProfileBasic } from './ProfileBasic'
import mockResponse from '../../../__mocks__/apiResponse.json'

afterEach(cleanup)

describe('ProfileBasic component', () => {
    const profile = mockResponse.results[0]

    test('should render with correct profile details', () => {
        const { asFragment } = render(
            <ProfileBasic
                picture={profile.picture.thumbnail}
                name={`${profile.name.first} ${profile.name.last}`}
                uuid={profile.login.uuid}
            />
        )
        expect(asFragment()).toMatchSnapshot()
    })

    test('should render with skeleton elements if no profile picture is provided', () => {
        const { asFragment } = render(
            <ProfileBasic
                name={`${profile.name.first} ${profile.name.last}`}
                uuid={profile.login.uuid}
            />
        )
        expect(asFragment()).toMatchSnapshot()
    })

    test('should render with skeleton elements if no profile name is provided', () => {
        const { asFragment } = render(
            <ProfileBasic
                picture={profile.picture.thumbnail}
                uuid={profile.login.uuid}
            />
        )
        expect(asFragment()).toMatchSnapshot()
    })

    test('should render with skeleton elements if no profile data is provided', () => {
        const { asFragment } = render(<ProfileBasic />)
        expect(asFragment()).toMatchSnapshot()
    })
})
