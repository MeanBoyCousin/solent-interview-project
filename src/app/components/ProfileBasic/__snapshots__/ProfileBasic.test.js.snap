// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileBasic component should render with correct profile details 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa QZupl"
    style="opacity: 0.2; transform: translateX(30px) translateZ(0);"
  >
    <img
      alt="The profile picture for Nella Aalto"
      class="thumbnail"
      src="https://randomuser.me/api/portraits/thumb/women/55.jpg"
    />
    <p>
      Nella Aalto
    </p>
    <img
      alt="An arrow pointing right enclosed inside a circle"
      class="arrow"
      src="/static/icons/circle-arrow.svg"
    />
  </div>
</DocumentFragment>
`;

exports[`ProfileBasic component should render with skeleton elements if no profile data is provided 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa QZupl"
    style="opacity: 0.2; transform: translateX(30px) translateZ(0);"
  >
    <div
      class="thumbnail skeleton skeleton-thumbnail"
      data-testid="skeleton"
    />
    <div
      class="skeleton skeleton-name"
    />
  </div>
</DocumentFragment>
`;
